src/sentry/static/sentry/app/actionCreators/organization.jsx

```diff
@@ -10,20 +10,29 @@ import ProjectsStore from 'app/stores/projectsStore';
  *
  * @param {Object} api A reference to the api client
  * @param {String} slug The organization slug
- * @param {boolean} detailed whether or not the detailed org details should be retrieved
+ * @param {boolean} detailed Whether or not the detailed org details should be
+ *                           retrieved
+ * @param {boolean} silent Should we silenty update the organization (do not
+ *                         clear the current organization in the store)
  */
-export async function fetchOrganizationDetails(api, slug, detailed) {
-  OrganizationActions.fetchOrg();
+export async function fetchOrganizationDetails(api, slug, detailed, silent) {
+  if (!silent) {
+    OrganizationActions.fetchOrg();
```

Comment:
My only concern might be to have the if statement further inside of the `OrganizationActions.fetchOrg` function. That way the store's state is reset to a loading state which should be correct, but just have an if statement surrounding the `this.trigger(this.get())` call inside of the org store. 

So changing the org store  reset() function to look like

```js
  reset(silent = false) {
    this.loading = true;
    this.error = null;
    this.errorType = null;
    this.organization = null;
    this.dirty = false;
    if (!silent) {
      this.trigger(this.get());
    }
  },
```
this way if the org details are chosen to be refetched and the request is not persisted in flight and immediately the user moves to the projects page or another page, the org details will still be refetched because it will detect that the org store is in a loading state. Let me know if I'm not making sense there.

Reply:
I can try this

Reply:
Talked with @billyvg, decided to keep the store state consistent with components.

Reply:
sounds good to me I was just scared of the case in which

1. user does something that requires a refetch of organization
2. trigger fetchOrganizationDetails()
3. orgstore is not reset
3. user clicks somewhere else unmounting and clearing the inflight request to `/organizations`
4. the org context sees that it does not need to refetch organization because the old org has not been reset
5. old information is displayed on the frontend

but maybe i have some misconceptions about how this works. I'd say if we run into problems we can just fix em on the fly and just go with what you have for now